proxy/src/serverless/backend.rs

```diff
@@ -42,7 +42,12 @@ impl PoolingBackend {
         };
 
         let secret = match cached_secret.value.clone() {
-            Some(secret) => secret,
+            Some(secret) => self.config.authentication_config.check_rate_limit(
```

Comment:
We also need to check for the endpoint rate limit, not only for auth.

Reply:
I think I want to save that for another PR. The endpoint rate limiter no longer provided much benefit now that we have endpoint role cache, so I don't feel strongly about enabling it and would probably rather remove the existing endpoint rate limiter